Notes link to iNat when recording Field Slip

JoeCohen opened the pull request:
- Adds a checkbox to the field_slip form. (See below.)
- Links notes[:Other_Codes] to iNat if the user checks that box
<img width="417" alt="Screenshot 2024-09-30 at 7 20 37 AM" src="https://github.com/user-attachments/assets/e5a03001-82cb-495f-8ef0-6148f4867ed1">
<img width="247" alt="Screenshot 2024-09-30 at 7 30 52 AM" src="https://github.com/user-attachments/assets/339e247c-a41c-40a6-a3f6-7ae36b8dfead">


### Manual Test
- http://localhost:3000/field_slips/qr_reader/new
- enter a valid Field Slip Code
- complete the form, including Location, ID, and a number in Other Codes,
- check the box
- Quick Create
Expected Result: It shows a new Observation whose Notes[:Other_Codes] is a link to an iNat observation.

JoeCohen committed:
Link notes[:Other_Codes] to iNat if user checks a box

- Adds a checkbox to the field_slip form
- Link notes[:Other_Codes] to iNat if the user checks that box

nimmolo commented on app/controllers/field_slips_controller.rb:
```diff
@@ -219,13 +226,9 @@ def update_notes_fields(notes)
-  def field_slip_id_by
-    user_str(params[:field_slip][:field_slip_id_by])
-  end
+  def field_slip_id_by = user_str(params[:field_slip][:field_slip_id_by])
```
This method `user_str` that parses `Jason Hollinger <jason>` to `Jason Hollinger` is repeated at several points in the code, including in pattern search. Can we please make this either
- A method of `User`
- A method of `StringExtensions`
and use that method everywhere in MO?

JoeCohen replied:
I agree ti would be good to keep things DRY, but I think it should be in a different PR.
I'm trying to keep this PR tiny so that there's a shot at getting it deployed in the next 24 hours.

JoeCohen committed:
Field Slip Record unassociated with Project iNat link

- Fixes Error thrown when user Quick Creates a Field Slip Record not associated with Project
